tests/Security/Http/EventListener/CheckTwoFactorCodeReuseListenerTest.php

```diff
@@ -0,0 +1,151 @@
+    {
+        $listener = new CheckTwoFactorCodeReuseListener($this->eventDispatcher, new stdClass(), 60, $this->logger);
+        $this->expectDoNothing();
+        $this->logger->expects($this->once())
+            ->method('error')
+            ->with(
+                'Your logger-cache seems to be configured wrongly! Provide a CacheItemPoolInterface '
+                .'as the cache object if you want to disallow reusing 2FA-codes!',
+            );
+
+        $listener->checkForCodeReuse(new TwoFactorCodeCheckEvent($this->user, self::MFA_CODE));
+    }
+
+    #[Test]
+    public function checkForCodeReuse_validCacheProviderAndNoHit_nothingHappens(): void
+    {
+        $cacheProvider = $this->createMock(CacheItemPoolInterface::class);
+        $cacheItem = new CacheItem();
+        $cacheItem->set(true);
+        $cacheItem->expiresAfter(60);
+        $cacheProvider->expects($this->once())
+            ->method('getItem')
+            ->with('scheb_two_factor_code_reuse.3468ccbd044e2fdfb0769b68b5c85d7660c6c12c')
+            ->willReturn($cacheItem);
+
+        $listener = new CheckTwoFactorCodeReuseListener($this->eventDispatcher, $cacheProvider, 20, $this->logger);
+        $this->expectDoNothing();
+
+        $listener->checkForCodeReuse(new TwoFactorCodeCheckEvent($this->user, self::MFA_CODE));
+    }
+
+    #[Test]
+    public function checkForCodeReuse_validCacheProviderWithCacheHit_eventIsDispatched(): void
+    {
+        $cacheProvider = $this->createMock(CacheItemPoolInterface::class);
+
+        $cacheItem = $this->createMock(CacheItemInterface::class);
+        $cacheItem->expects($this->once())->method('isHit')->willReturn(true);
+
+        $cacheProvider->expects($this->once())
+            ->method('getItem')
+            ->willReturn($cacheItem);
+
+        $this->eventDispatcher->expects($this->once())
+            ->method('dispatch')
+            ->with(new TwoFactorCodeReusedEvent($this->user, self::MFA_CODE));
+
+        $listener = new CheckTwoFactorCodeReuseListener($this->eventDispatcher, $cacheProvider, 60, null);
+
+        $listener->checkForCodeReuse(new TwoFactorCodeCheckEvent($this->user, self::MFA_CODE));
+    }
+
+    #[Test]
+    public function checkForCodeREuse_validCacheProviderNoCachewHit_CacheItemIsSaved(): void
+    {
+        $cacheProvider = $this->createMock(CacheItemPoolInterface::class);
+        $cacheProvider->expects($this->once())
+            ->method('save')
+            ->willReturnCallback(function (CacheItem $item) {
+                $this->assertFalse($item->isHit());
+                $this->assertSame(true, $item->get());
+                $sweetsThief = Closure::bind(function (CacheItem $item) {
+                    return $item->expiry;
+                }, null, CacheItem::class);
+                $this->assertGreaterThanOrEqual(time() + 20, $sweetsThief($item));
+                $this->assertLessThanOrEqual(time() + 21, $sweetsThief($item));
```

Comment:
I didn't find an easier way to handle this as the correct lifetime should be checked here. And getting that from the CacheItem is only possible by either creating an own implementation of the CacheItemInterface or using the existing one. Using the existing one in combination with this abomination was less code, so I went that route. 

Can change that though if you think it'd be better for future maintenance.